keras_nlp/samplers/sampler.py

```diff
@@ -39,6 +40,17 @@
     from_logits: bool, defaults to True. Indicate if the `token_probability_fn`
         returns logits. If False, `token_probability_fn` returns probability
         distributions.
+    cache: a dense int tensor, the cache used in decoding. The cache
+        stores the K and V of each
+        `keras_nlp.layers.CachedMultiHeadAttention` layer to make the
+        decoding faster by avoiding duplicated computation. See more
+        details in the docstring of
+        `keras_nlp.layers.CachedMultiHeadAttention`.
+    token_probs: a dense float tensor of shape
```

Comment:
This still feels pretty confusing as an arg.

As a user, I would think. We have a token_probability_fn and a token_probs pass in? What is the point of passing a function for compute the next tokens if I also have to pass them in.

I'm pretty sure we can avoid adding this arg entirely, will play around.

Reply:
talked offline, let's remove it for now, if contrastive search needs it, let's add it back